Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/ObjectManipulatorMigrationHandler.cs

```diff
@@ -23,82 +23,172 @@ public bool CanMigrate(GameObject gameObject)
         /// <inheritdoc />
         public void Migrate(GameObject gameObject)
         {
-            var mh1 = gameObject.GetComponent<ManipulationHandler>();
-            var mh2 = gameObject.AddComponent<ObjectManipulator>();
+            var manipHandler = gameObject.GetComponent<ManipulationHandler>();
+            var objManip = gameObject.AddComponent<ObjectManipulator>();
 
-            mh2.HostTransform = mh1.HostTransform;
+            objManip.HostTransform = manipHandler.HostTransform;
 
-            switch (mh1.ManipulationType)
+            switch (manipHandler.ManipulationType)
             {
                 case ManipulationHandler.HandMovementType.OneHandedOnly:
-                    mh2.ManipulationType = ObjectManipulator.HandMovementType.OneHanded;
+                    objManip.ManipulationType = ManipulationHandFlags.OneHanded;
                     break;
                 case ManipulationHandler.HandMovementType.TwoHandedOnly:
-                    mh2.ManipulationType = ObjectManipulator.HandMovementType.TwoHanded;
+                    objManip.ManipulationType = ManipulationHandFlags.TwoHanded;
                     break;
                 case ManipulationHandler.HandMovementType.OneAndTwoHanded:
-                    mh2.ManipulationType = ObjectManipulator.HandMovementType.OneHanded |
-                        ObjectManipulator.HandMovementType.TwoHanded;
+                    objManip.ManipulationType = ManipulationHandFlags.OneHanded |
+                        ManipulationHandFlags.TwoHanded;
                     break;
             }
 
-            mh2.AllowFarManipulation = mh1.AllowFarManipulation;
-            mh2.OneHandRotationModeNear = (ObjectManipulator.RotateInOneHandType)mh1.OneHandRotationModeNear;
-            mh2.OneHandRotationModeFar = (ObjectManipulator.RotateInOneHandType)mh1.OneHandRotationModeFar;
+            objManip.AllowFarManipulation = manipHandler.AllowFarManipulation;
 
-            switch (mh1.TwoHandedManipulationType)
+            if (manipHandler.OneHandRotationModeNear == manipHandler.OneHandRotationModeFar)
+            {
+                MigrateOneHandRotationModes(ref objManip, manipHandler.OneHandRotationModeNear, ManipulationProximityFlags.Near | ManipulationProximityFlags.Far);
+            }
+            else
+            {
+                MigrateOneHandRotationModes(ref objManip, manipHandler.OneHandRotationModeNear, ManipulationProximityFlags.Near);
+                MigrateOneHandRotationModes(ref objManip, manipHandler.OneHandRotationModeFar, ManipulationProximityFlags.Far);
+            }
```

Comment:
Out of curiousity, if OneHandRotationModeNear == OneHandRotationModeFar, would it still have the same effect to have just these two lines of code?

```c#
MigrateOneHandRotationModes(ref objManip, manipHandler.OneHandRotationModeNear, ManipulationProximityFlags.Near);
MigrateOneHandRotationModes(ref objManip, manipHandler.OneHandRotationModeFar, ManipulationProximityFlags.Far);
```

Just looking to see places in the code where we can reduce complexity.

Reply:
Not quite. It would produce the same behaviour, but it could potentially add two constraint components of the same type where one constraints near and the other constrains far. I think it's preferable to only add one constraint component in this case.